packages/devtools_app/lib/src/screens/debugger/debugger_screen.dart

```diff
@@ -199,9 +200,13 @@ class DebuggerScreenBodyState extends State<DebuggerScreenBody>
   void _onNodeSelected(VMServiceObjectNode? node) {
     final location = node?.location;
     if (location != null) {
-      controller.codeViewController.showScriptLocation(
-        location,
-        focusLine: true,
+      final routerDelegate = DevToolsRouterDelegate.of(context);
+      routerDelegate.updateStateIfNotCurrent(
```

Comment:
just piecing this new system together: does the debugger have to listen for router updates somewhere?

Reply:
It can, but in this situation, it doesn't need to since the `CodeViewController` is handling the state update. For example, if we wanted to support history for debugger frame selection (not that I think we should do this), the `DebuggerController` would need to be able to listen to router updates to determine if it should update the currently selected frame.